src/dataset.rs

```diff
@@ -305,6 +305,102 @@ impl Dataset {
         }
         Ok(transformation)
     }
+
+    /// For datasources which support transactions, this creates a transaction.
+    ///
+    /// If starting the transaction fails, will return [`OGRErr::OGRERR_FAILURE`]. Datasources
+    /// which do not support efficient transactions will always return
+    /// [`OGRErr::OGRERR_UNSUPPORTED_OPERATION`].
+    ///
+    /// Some drivers may offer an emulation of transactions, but sometimes with significant
+    /// overhead. Use [`try_start_transaction`](Dataset::try_start_transaction) if you only want to
+    /// start a transaction if it can be done efficiently.
+    ///
+    /// Nested transactions are not supported.
+    ///
+    /// All changes done after the start of the transaction are applied to the datasource when
+    /// [`commit`](Transaction::commit) is called. They may be canceled by calling
+    /// [`rollback`](Transaction::rollback) instead.
+    ///
+    /// At the time of writing, transactions only apply on vector layers.
+    ///
+    /// Datasets that support transactions will advertise the ODsCTransactions capability. Use of
+    /// transactions at dataset level is generally preferred to transactions at layer level, whose
+    /// scope is rarely limited to the layer from which it was started.
+    ///
+    /// If an error occurs after a successful `start_transaction`, the whole transaction may or may
+    /// not be implicitly canceled, depending on the driver. For example, the PG driver will cancel
+    /// it, but the SQLite and GPKG drivers will not.
```

Comment:
I'm surprised this is true for SQLite, which, in general supports transaction rollback. Can you provide any more detail?

Reply:
I just copied most of it from the GDAL API docs (with minor wordsmithing) so I have no clue either.